pkg/services/internal/whatsonchain/script_history.go

```diff
@@ -0,0 +1,144 @@
+package whatsonchain
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+)
+
+func validateScriptHash(scriptHash string) error {
+	if scriptHash == "" {
+		return fmt.Errorf("scripthash cannot be empty")
+	}
+
+	if len(scriptHash) < 20 {
+		return fmt.Errorf("invalid scripthash length: too short (minimum 20 characters)")
+	}
+
+	if len(scriptHash) > 66 {
+		return fmt.Errorf("invalid scripthash length: too long (maximum 66 characters)")
+	}
+
+	_, err := hex.DecodeString(scriptHash)
+	if err != nil {
+		return fmt.Errorf("invalid scripthash format: %w", err)
+	}
+
+	return nil
+}
+
+func (woc *WhatsOnChain) getUnconfirmedScriptHistory(ctx context.Context, scriptHash string) (*wdk.ScriptHistoryResult, error) {
+	var history wdk.ScriptHashHistoryResponse
+	url := fmt.Sprintf("%s/script/%s/unconfirmed/history", woc.url, scriptHash)
+
+	res, err := woc.httpClient.R().
+		SetContext(ctx).
+		SetResult(&history).
+		Get(url)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to get unconfirmed script history: %w", err)
+	}
+
+	if res.StatusCode() != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code %d getting unconfirmed script history", res.StatusCode())
+	}
+
+	if history.Error != "" {
+		return nil, fmt.Errorf("API error: %s", history.Error)
+	}
+
+	historyItems := make([]wdk.ScriptHistoryItem, len(history.Result))
+	for i, item := range history.Result {
+		historyItems[i] = wdk.ScriptHistoryItem{
+			TxHash: item.TxID,
+			Height: item.Height,
+		}
+	}
```

Comment:
You could create function:
```go
func toScriptHistoryItem(item wdk.ScriptHashHistoryItem) wdk.ScriptHistoryItem {
	return wdk.ScriptHistoryItem{
		TxHash: item.TxID,
		Height: item.Height,
	}
}
```

and then 

```suggestion
	historyItems := slices.Map(history.Result, toScriptHistoryItem)
```

Reply:
function created, suggestion implemented